### PR TITLE
PAR-40 Improvements

### PR DESCRIPTION
**Pull request by stemount**
We will need to work out a compromise for this table in GDS theme!!

image

### CODE REVIEW

File: web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+
+    $options['status_fields'] = ['default' => []];
+    $options['status_on_label'] = ['default' => 'Confirmed'];
```
Comment (kalpaitch) [inline web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php]: Sorry default values shouldn't be defined here. We gotta try and ready them only from the yaml config..

File: web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    parent::buildOptionsForm($form, $form_state);
+
+    // @todo figure out how to get the label from the entity configuration.
+    $form['status_fields'] = [
+      '#title' => 'Status fields to include',
+      '#description' => 'Select multiple PAR Partnership status fields',
+      '#type' => 'checkboxes',
+      '#options' => [
+        'partnership_info_agreed_authority' => 'PA Partnership Info Agreed',
+        'partnership_info_agreed_business' => 'Business Partnership Details Confirmed',
```
Comment (kalpaitch) [inline web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php]: and again, values should come from config.

File: web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::buildOptionsForm($form, $form_state);
+
+    // @todo figure out how to get the label from the entity configuration.
```
Comment (stemount) [inline web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php]: heh, @kalpaitch :)

Comment (kalpaitch) [inline web/modules/custom/par_data/src/Plugin/views/field/ParPartnershipsCombinedStatusField.php]: ha

File: web/modules/custom/par_data/par_data.views.inc [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * Implements hook_views_data_alter().
+ */
+function par_data_views_data_alter(array &$data) {
```
Comment (stemount) [inline web/modules/custom/par_data/par_data.views.inc]: Yes, this will be in ViewData. thx. :)

**Comment by stemount** [comment]
ta @nao360 this will save the table overflowing.

**Pull request merged**